core/variable_binary_arithmetic.cpp

```diff
@@ -147,11 +158,43 @@ template <class T1, class T2> Variable or_op(const T1 &a, const T2 &b) {
       overloaded{[](const auto &var_, const auto &other_) -> bool {
                    return var_ | other_;
                  },
-                 [](const units::Unit &aUnit, const units::Unit &bUnit) {
-                   expect::equals(aUnit, units::dimensionless);
-                   expect::equals(bUnit, units::dimensionless);
-                   return aUnit;
-                 }});
+                 dimensionless_unit_check_return});
+}
+
+template <class T1, class T2> T1 &and_equals(T1 &variable, const T2 &other) {
```

Comment:
Can you unify all of the implementation of binary operations, similar to how done above for, e.g., `plus`, avoiding a lot of duplication?

Reply:
Do you mean move the lambda (e.g. `[](const auto &var_, const auto &other_) -> bool { return var_ | other_; }`) to `static constexpr auto`?

Reply:
Yes (but I realize now that it does not save so much). But remember the discussion on Slack recently regarding instantiated code size for lambdas in templated functions?